src/manifolds/SymplecticStiefel.jl

```diff
@@ -0,0 +1,554 @@
+
+Given a point ``p \in \operatorname{SpSt}(2n, 2k)``, every tangent vector
+``X \in T_p\operatorname{SpSt}(2n, 2k)`` is of the form
+``X = \tilde{\Omega}p``, with
+````math
+    \tilde{\Omega} = \left(I_{2n} - \frac{1}{2}pp^+\right)Xp^+ -
+                     pX^+\left(I_{2n} - \frac{1}{2}pp^+\right) \in ℝ^{2n \times 2n},
+````
+as shown in Proposition 3.5 of [^Bendokat2021].
+Using this representation of ``X``, the Cayley retraction
+on ``\operatorname{SpSt}(2n, 2k)`` is defined pointwise as
+````math
+    \mathcal{R}_p(X) = \operatorname{cay}\left(\frac{1}{2}\tilde{\Omega}\right)p.
+````
+The operator ``\operatorname{cay}(A) = (I - A)^{-1}(I + A)`` is the Cayley transform.
+
+However, the computation of an ``2n \times 2n`` matrix inverse in the expression
+above can be reduced down to inverting a ``2k \times 2k`` matrix due to Proposition
+5.2 of [^Bendokat2021].
+
+Let ``A = p^+X`` and ``H = X - pA``. Then an equivalent expression for the Cayley
+retraction defined pointwise above is
+````math
+    \mathcal{R}_p(X) = -p + (H + 2p)(H^+H/4 - A/2 + I_{2k})^{-1}.
+````
+It is this expression we compute inplace of `q`.
+"""
+function retract!(M::SymplecticStiefel{n,k}, q, p, X, ::CayleyRetraction) where {n,k}
+    # Define intermediate matrices for later use:
+    A = symplectic_inverse_times(M, p, X)
+
+    H = X .- p * A  # Allocates (2n × 2k).
+
+    # A = I - A/2 + H^{+}H/4:
+    A .= (symplectic_inverse_times(M, H, H) ./ 4) .- (A ./ 2)
+    Manifolds.add_scaled_I!(A, 1.0)
+
+    # Reuse 'H' memory:
+    H .= H .+ 2 .* p
+    r = lu!(A)
+    q .= (-).(p) .+ rdiv!(H, r)
+    return q
+end
+
+@doc raw"""
+    inverse_retract!(::SymplecticStiefel, q, p, X, ::CayleyInverseRetraction)
+
+Compute the Cayley Inverse Retraction ``X = \mathcal{L}_p^{\operatorname{SpSt}}(q)``
+such that the Cayley Retraction from ``p`` along ``X`` lands at ``q``, i.e.
+``\mathcal{R}_p(X) = q`` [^Bendokat2021].
+
+First, recall the definition the standard symplectic matrix
+````math
+Q =
+\begin{bmatrix}
+ 0    &  I \\
+-I    &  0
+\end{bmatrix}
+````
+as well as the symplectic inverse of a matrix ``A``, ``A^{+} = Q^T A^T Q``.
+
+For ``p, q ∈ \operatorname{SpSt}(2n, 2k, ℝ)`` then, we can define the
+inverse cayley retraction as long as the following matrices exist.
+````math
+    U = (I + p^+ q)^{-1} \in ℝ^{2k \times 2k},
+    \quad
+    V = (I + q^+ p)^{-1} \in ℝ^{2k \times 2k}.
+````
+
+If that is the case, the inverse cayley retration at ``p`` applied to ``q`` is
+````math
+\mathcal{L}_p^{\operatorname{Sp}}(q) = 2p\bigl(V - U\bigr) + 2\bigl((p + q)U - p\bigr)
+                                        ∈ T_p\operatorname{Sp}(2n).
+````
+
+[^Bendokat2021]:
+    > Bendokat, Thomas and Zimmermann, Ralf:
+	> The real symplectic Stiefel and Grassmann manifolds: metrics, geodesics and applications
+	> arXiv preprint arXiv:2108.12447, 2021 (https://arxiv.org/abs/2108.12447)
+"""
+function inverse_retract!(M::SymplecticStiefel, X, p, q, ::CayleyInverseRetraction)
```

Comment:
We usually document the `inverse_retract` variant.

Reply:
Done.